crypto/evp/evp_fetch.c

```diff
@@ -132,10 +132,13 @@ static int put_method_in_store(OPENSSL_CTX *libctx, void *store,
         && (store = get_default_method_store(libctx)) == NULL)
         return 0;
 
-    if (methdata->refcnt_up_method(method)
-        && ossl_method_store_add(store, methid, propdef, method,
-                                 methdata->destruct_method))
-        return 1;
+    if (methdata->refcnt_up_method(method)) {
+        if (ossl_method_store_add(store, methid, propdef, method,
```

Comment:
I kind of wonder, if this duplicate detection should be done on the ossl_method_store_add
where we have a lock for that?

Reply:
That's a refactoring question that I'd rather keep out of this PR.

Reply:
I start to think that the duplicate detection should be in `ossl_method_store_add`
and not in `ossl_method_construct_this`.
Why should the store add the same object twice?  Is there a use case for it?

Reply:
The "local" store is *temporary*